instrumentation/tomcat/tomcat-jdbc/src/test/java/io/opentelemetry/javaagent/instrumentation/tomcat/jdbc/TomcatJdbcInstrumentationTest.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.tomcat.jdbc;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.BDDMockito.given;
+
+import io.opentelemetry.instrumentation.testing.junit.AgentInstrumentationExtension;
+import io.opentelemetry.instrumentation.testing.junit.db.DbConnectionPoolMetricsAssertions;
+import java.sql.Connection;
+import org.apache.tomcat.jdbc.pool.DataSource;
+import org.assertj.core.api.AbstractIterableAssert;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.api.extension.RegisterExtension;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+public class TomcatJdbcInstrumentationTest {
+
+  @RegisterExtension
+  static final AgentInstrumentationExtension testing = AgentInstrumentationExtension.create();
+
+  @Mock javax.sql.DataSource dataSourceMock;
+  @Mock Connection connectionMock;
+
+  @Test
+  void shouldReportMetrics() throws Exception {
+    // given
+    given(dataSourceMock.getConnection()).willReturn(connectionMock);
+
+    DataSource tomcatDataSource = new DataSource();
+    tomcatDataSource.setDataSource(dataSourceMock);
+
+    // there shouldn't be any problems if this methods gets called more than once
+    tomcatDataSource.createPool();
+    tomcatDataSource.createPool();
+
+    // when
+    Connection connection = tomcatDataSource.getConnection();
+    connection.close();
+
+    // then
+    assertConnectionPoolMetrics(tomcatDataSource.getPoolName());
+
+    // when
+    // this one too shouldn't cause any problems when called more than once
+    tomcatDataSource.close();
+    tomcatDataSource.close();
+    testing.clearData();
+
+    // then
+    assertNoConnectionPoolMetrics();
```

Comment:
do we need to wait at least [metric interval] before this assertion?

also having similar conversation https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/6099#discussion_r888266657